### PR TITLE
Update stlSerializer.ts

### PR DESCRIPTION
**Pull request by aWeirdo**
Fix wrong orientation, flipped faces and bloated binary export.

-- Original mesh
head_original

-- After current export
head_export_now

-- After fixed export
head_export_fixed

### CODE REVIEW

File: serializers/src/stl/stlSerializer.ts [hunk 1]
```diff
@@ -12,23 +12,23 @@ export class STLExport {
...
-                new Vector3(vertices[id[1]], vertices[id[1] + 1], vertices[id[1] + 2]),
-                new Vector3(vertices[id[2]], vertices[id[2] + 1], vertices[id[2] + 2])
+                new Vector3(vertices[id[0]], vertices[id[0] + 2], vertices[id[0] + 1]),
+                new Vector3(vertices[id[1]], vertices[id[1] + 2], vertices[id[1] + 1]),
+                new Vector3(vertices[id[2]], vertices[id[2] + 2], vertices[id[2] + 1])
             ];
             let p1p2 = v[0].subtract(v[1]);
             let p3p2 = v[2].subtract(v[1]);
-            let n = (Vector3.Cross(p1p2, p3p2)).normalize();
+            let n = (Vector3.Cross(p1p2, p3p2)).normalize().negateInPlace();
```
Comment (RaananW) [inline serializers/src/stl/stlSerializer.ts:31]: Just wondering about the normal negation - why is this needed?

**Review by RaananW — APPROVED** [review]

File: serializers/src/stl/stlSerializer.ts (hunk 1 above)
Comment (Popov72) [reply serializers/src/stl/stlSerializer.ts:31]: It must be because the vertices are now x/z/y instead of x/y/z.

But it may be easier to simply do: `let n = (Vector3.Cross(p3p2, p1p2)).normalize()`

**Comment by aWeirdo** [comment]
Last minute changes,
Removed n vector3 negating & switched the input values as suggested.
Updated default fileName to match binary mesh name (which is always 'stlmesh')
Updated ASCII mesh name to match binary mesh name for consistency
And removed fileName check in download code as it is defaulted at beginning.

**Review by RaananW — APPROVED** [review]
